src/openforms/submissions/models/submission.py

```diff
@@ -460,6 +462,10 @@ def render_confirmation_page(self) -> str:
             "_form": self.form,  # should be the same as self.form
             "public_reference": self.public_registration_reference,
             **self.data,
+            **{
+                variable.key: variable.initial_value
+                for variable in get_static_variables(submission=self)
+            },
```

Comment:
note that the order matters here! now the static variabels overwrite any variables possibly added by form fields. This shouldn't happen due to the unique key validation and checks that you're not using static variables, but it is possible new static variables in the future cause problems with existing forms.

I'm also surprised this happened in the first place - clearly we're building template context differently in different places. So I'd rather see a refactor/cleanup that defines a function `get_variables_for_context(submission)` somewhere that encapsulates this for anything using the django template language for rendering (confirmation text, confirmation e-mail, registration e-mail, dynamic formio labels/etc.).